### PR TITLE
devise functionality done.

### PR DESCRIPTION
**Pull request by DilsherB**
Hello. Kindly review my request to merge devise branch to development branch.
The branch contains the implementation of the following features.

1. User can register a new user.
2. User logs in with a combination of email and password.
3. Hashed password should be stored in the database.
4. Ask for confirmation of email.
5. Can reset password.

### CODE REVIEW

File: app/views/devise/sessions/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  <% end %>
+
+  <div class="actions">
+    <%= f.submit "Log in", class: 'button margintop20' %>
+  </div>
+<% end %>
+
+<%= render "devise/shared/links" %>
+
+
```
Comment (julie-ify) [inline app/views/devise/sessions/new.html.erb]: - [ ] Kindly follow the wireframe for your page layout. Your login form can be improved. We need to extend the forms to have enough width on the screen and the elements should be centered, not left-aligned to a centered container.

  image

File: app/views/devise/registrations/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    <%= f.label :password_confirmation %><br />
+    <%= f.password_field :password_confirmation, autocomplete: "new-password" %>
+  </div>
+
+  <div class="actions">
+    <%= f.submit "Sign up", class: 'button margintop20' %>
+  </div>
+<% end %>
+
+<%= render "devise/shared/links" %>
```
Comment (julie-ify) [inline app/views/devise/registrations/new.html.erb]: - [ ] Kindly center the sign up form too.

  image

File: app/models/user.rb [hunk 3]
```diff
@@ -1,4 +1,8 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
```
Comment (julie-ify) [inline app/models/user.rb:5]: - [ ] When a user signs up the app logs them in at once. Kindly note that when I sign up as a user, I should be required to confirm my email before I can log in. Kindly add the `:confirmable` config to the user devise setup.

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @DilsherB and @adarachel,

Good job so far!

## Highlights!
- Good code organization and working devise ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mukhammadrizooff — APPROVED** [review]
Hi @TEAM, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

 

## Highlights
- No linter errors ✔️ 
- Correct GitFlow ✔️ 
- Good implementations ✔️
- Descriptive `README` file ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**